### DISCUSSION TITLE
Quick Veto Temp Time ?

### DISCUSSION THREAD

**Discussion by boskape** [john30/ebusd#1261] [Q&A]
Hi
Ich habe ein Vailant VRC720f/2 Funk Raum Thermostat
dies wird als BASV Device erkannt

Bei dem Thermostat kann man für eine gewisste Zeit eine Wunsch Temperatur setzen und dann heizt die Therme auf diese Temp für  die eingestellte Zeit

die Quick VetoTemp hab ich
ebusd basv z1QuickVetoTemp tempv

aber es gibt keine Veto Time
Wisst Ihr eventuell was da fehlt
oder wie ich das im HA nutzen kann

Danke

**Comment by tttooommmeeekkk** [comment]
Ich habe zwar kein solches Funk-Termostat aber eine an meine WP per Bus-Kabel angebundene SonsoComfort-Steuer-Einheit (bei mir als CTLV2-Device erkannt). Die Steuerungsmöglichkeiten sind wahrscheinlich vergleichbar?!

In den Konfigurationsdateien von rmalbrecht ist zumindest die `z1QuickVetoEndTime` (für die weiteren Zonen analog z2 und z3, die bei mir nicht aktiv sind, da ich nur eine Heizzone hab) zu finden (man kann damit die Endzeit übersteuern; aber eben nicht direkt die Laufzeit der "Zwischenheizen-Steuerung" eingeben, wie sie am SensoComfort-Regler in 0,5h-Schritten bis 12h wählbar ist). Habe meine lokale csv-Datei entsprechend angepasst (r;w), dass der Wert auch schreibbar ist.
Vielleicht hilft das. Zurückgegeben wurde mir der Wert in HA allerdings nicht (habe die angepasste Endzeit lediglich per `ebusctl` oder am SensoComfort-Regler ablesen können).

Nebenbei (falls von Interesse bzw. Steuerung per HA als verfolgtes Ziel):
Ein Starten des "Veto-Heizens" habe ich per HA allerdings nicht hinbekommen (trotz erfolgreichen Setzens der Werte `z1QuickVetoTemp`, `z1SFMode`, `z1QuickVetoEndTime` und `z1ActualHeatingRoomTempDesired`); wahrscheinlich fehlt da noch etwas. Auslesen/Mitlesen der manuellen Steuerung per `ebusctl grab` ... habe ich versucht, aber um die Resultate korrekt zu deuten, dazu fehlt mir das Verständnis.
Ich habe die Veto-Steuerung per HA daher aktuell  erst mal auf Eis gelegt und verfolge eine andere Strategie:

Ich werde jetzt die Sensoren `z1OpModeHeating` und `z1HeatingRoomTempDesiredManualControlled` und die analogen Sensoren für die Kühlung nutzen, um eine spontane Heiz-/Kühl-Anforderung per HA umzusetzen bzw. ggf. bei Bedarf auch zu automatisieren.

Hier noch der link zu den o.g. Dateien:
https://github.com/rmalbrecht/ebusd-configuration/blob/master/ebusd-2.1.x/de/vaillant/15.ctlv2.SW0514.HW1104.csv
Webseite mit entsprechendem BLOG-Eintragen (aus Dez 2023 - zwischenzeitlich wurde an den csv weitergearbeitet):
https://romal.de/2023/12/23/ebusd-vaillant-waermepumpe-und-home-assistant/

**Reply by boskape** [reply]
HAllo

Danke für die Erklärung
Ich hab schon mal versucht eine andere csv datei unter zu jubeln aber es wurde dann meine Hardware nicht erkannt
bei mir wählt der ebusd dann immer die BASV.csv

möglicherweise liegt sie am falschen Platz
wie hast Du das bei dir abgelegt wenn ich fragen darf

Wenn ich das dann mal hab wäre die nächste Hürde es mit HA zum laufen zu bringen

Möglicherweise hilf da der Beitrag
https://community.home-assistant.io/t/ebusd-vaillant-heating-ideas-to-optimize-self-consumption/540632/7

**Reply by chrizzzp** [reply]
Wir haben das in diesem PR kürzlich herausgefunden:  https://github.com/stadid/ebusd-configuration/pull/2

Die Erkenntnisse finden sich alle hier wieder: https://github.com/jonesPD/ebusd-configuration/blob/master/ebusd-2.1.x/en/vaillant/15.ctlv2.csv

>   1. write desired temperature to `z*QuickVetoTemp`
 >  2. write desired mode duration in hours to `z*QuickVetoDuration` (valid values are 0.5-12 with 0.5 increments)

```
r;w,,z1QuickVetoTemp,Quick Veto Temperature Zone 1,,,,0800,,,tempv,,,manual override setpoint for zone 1
r;w,,z1QuickVetoDuration,Duration of Quick Veto mode in Hours Zone 1,,,,2600,,,EXP,,,valid values are 0.5-12 in 0.5 increments (default 3) writing enables quick veto mode
```

**Reply by boskape** [reply]
OK Danke
aber darf ich fragen wo ihr die CSV ablegt beim HA
ich habe es versucht unter etc/ebusd oder im Config Folder von HA, Config/Ebusd um sie auch aus HA mit dem File Editor zu bearbeiten
Aber irgendwie nimmt es immer als BASV erkannt

Daher hab ich wieder auf die Online Version verwiesen
Danke euch für die Unterstützung

**Reply by chrizzzp** [reply]
Wie ist ebusd in HA eingebunden? Über das ebusd-Addon? Falls ja, dann kann man im Addon auf der Konfigurationsseite einstellen (Config path), wo die lokalen CSV-Dateien liegen, üblicherweise in einem eigenen Unterverzeichnis des Config-Verzeichnis von HA. 

Man muß aber alle CSVs dann lokal vorliegen haben (also am besten geklont vom offiziellen Online-Reposititory). Dann kann man beginnen die lokalen CSV-Dateien zu modifizieren. 

Die 15.basv.csv verweist nur auf die 15.700.csv, deshalb würde ich die 15.ctlv2.csv aus dem o.g. Repo (https://github.com/jonesPD/ebusd-configuration/blob/master/ebusd-2.1.x/en/vaillant/15.ctlv2.csv) runterladen und lokal in 15.basv.csv umbenennen.

**Reply by tttooommmeeekkk** [reply]
@chrizzzp 
Großartig, danke für den link auf das "Projekt". Die `z*QuickVetoDuration` fehlte mir noch in meiner "Raupensammlung". Werde das  bei Gelegenheit vielleicht noch mal in meine lokale csv einbauen und die originale Quick-Veto-Funktion testen. Aktuell passt die Nutzung des manuellen Modus  eigentlich auch ganz gut. Hab mir nen Uhrzeit-Helfer in HA gebastelt und den zum Zurückschalten in eine Automation genutzt (die man ja aktivieren oder deaktivieren kann). So kann man die Veto-Funktion ja auch "nachbauen".
So langsam lässt sich die Anlage endlich über den ebusd und HA einigermaßen komfortabel steuern ohne dieses Cloud-Gedöns.

@boskape 
edit: --gelöscht-- (weil wenig hilfreich und zeitl. Antwort-Überschneidung)

**Reply by boskape** [reply]
@chrizzzp 
Ja ist über das ebusd-addon eingebunden
ICh werde das nochmal probieren die CSV lokal abzulegen

reicht es die CSV umzubenennen
Wird das nicht aufgrund der Hardware ID erkannt

**Reply by chrizzzp** [reply]
Korrekt, die Hardware-ID bestimmt welche CSV-Datei geladen wird. Die Hardware-IDs zeigt Dir z.B.
`ebusctl info` auf Kommandozeilenebene (die aber im ebusd Docker-Container laufen muss) oder (im Fall für den Regler) die HA-Entität `sensor.ebusd_scan_15_id`. Umbenennen der CSV-Dateien reicht, es sind nur einfache Textdateien mit komma-getrennten Werten.
